kodilla-patterns2/src/main/java/com/kodilla/patterns2/adapter/bookclasifier/adapter/bookclassifier/MedianaAdapter.java

```diff
@@ -3,24 +3,33 @@
 import com.kodilla.patterns2.adapter.bookclasifier.librarya.Book;
 import com.kodilla.patterns2.adapter.bookclasifier.librarya.Classifier;
 import com.kodilla.patterns2.adapter.bookclasifier.libraryb.BookSignature;
+import com.kodilla.patterns2.adapter.bookclasifier.libraryb.Statistics;
 
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 
 public class MedianaAdapter extends MedianAdaptee implements Classifier {
 
+    public MedianaAdapter(Statistics statistics) {
+        super(statistics);
+    }
+
     @Override
     public int publicationYearMedian(final Set<Book> bookSet) {
 
         Map<BookSignature, com.kodilla.patterns2.adapter.bookclasifier.libraryb.Book> books = new HashMap<>();
-        for (Book book: bookSet) {
-           books.put(new BookSignature(book.getSignature()),
-                   new com.kodilla.patterns2.adapter.bookclasifier.libraryb.Book
-                           (book.getAuthor(), book.getTitle(), book.getPublicationYear())
-           );
-        }
-       return medianPublicationYear(books);
+        bookSet.forEach(b -> books.put(createBookSignatureForLibraryB(b), createBookLibraryBType(b)));
+        return medianPublicationYear(books);
+    }
+
+    private com.kodilla.patterns2.adapter.bookclasifier.libraryb.Book createBookLibraryBType(Book book) {
```

Comment:
skomplikowane te nazwy coś. W obrębie danej klasy, jak metody są prywatne w szczególności, to staramy się używać jak najprostszych i najpełniej mówiących zarazem nazw. Moim zdaniem (taka mała sugestia) nazwałbym metody convertBook() oraz createSignature() :D